src/Pages/AddNewRoom/AddNewRoom.tsx

```diff
@@ -0,0 +1,178 @@
+import { Button, MenuItem, TextField } from '@mui/material'
+import Box from '@mui/material/Box';
+import { useEffect, useState } from 'react'
+import './AddNewRoom.module.scss'
+import CloudUploadIcon from '@mui/icons-material/CloudUpload';
+import { useDispatch } from 'react-redux';
+import { FacilitiesData } from '@/Redux/Features/Facilities/FacilitiesSlice';
+import { useForm } from 'react-hook-form';
+import { ChevronRight } from '@mui/icons-material';
+import { CreateRooms } from '@/Redux/Features/Rooms/CreateRoomsSlice';
+import { Link, useNavigate } from 'react-router-dom';
+import { toast } from "react-toastify";
+const AddNewRoom = () => {
+  const {
+    register,
+    handleSubmit,
+    getValues,
+    formState: { errors },
+  } = useForm();
+
+  const navigate = useNavigate()
+
+  const required = "This Field is required"
+
+  //? ***************Get Facilities Data ***************
+  const dispatch = useDispatch();
+  const [selectData, setSelectData] = useState(null)
+
+
+  const getFacilitiesData = async () => {
```

Comment:
consider making custom hooks for repeated data fetching, for example the facilities will be required in rooms, add-room, facilities page.
You can make useFacilities to abstract the logic for it and reuse it in other pages.
The same goes for any logic that will be repeated.